applications/GeoMechanicsApplication/tests/cpp_tests/test_thermal_dispersion_law.cpp

```diff
@@ -0,0 +1,98 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//                   Gennady Markelov
+//
+
+#include "custom_constitutive/thermal_dispersion_2D_law.hpp"
+#include "geo_mechanics_application.h"
+#include "includes/ublas_interface.h"
+#include "testing/testing.h"
+
+namespace Kratos::Testing {
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateThermalDispersionMatrix2D, KratosGeoMechanicsFastSuite)
+{
+    Model current_model;
+    auto& r_model_part = current_model.CreateModelPart("ModelPart", 1);
+
+    auto cond_prop = r_model_part.CreateNewProperties(0);
+    cond_prop->SetValue(POROSITY, 0.5);
+    cond_prop->SetValue(SATURATION, 0.75);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_WATER, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XX, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XY, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YY, 1000.0);
+
+    Matrix ThermalDispersionMatrix = ZeroMatrix(2, 2);
```

Comment:
Since it's a local variable, it should be lower_case
```suggestion
    Matrix thermal_dispersion_matrix = ZeroMatrix(2, 2);
```